waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/impl/StaticDatabaseMappingServiceTest.java

```diff
@@ -65,22 +65,26 @@
   private static final String URI = "uri";
 
   private @Mock MetaStoreMappingFactory metaStoreMappingFactory;
+  private @Mock Iface primaryDatabaseClient;
+  private @Mock Iface federatedDatabaseClient;
 
   private StaticDatabaseMappingService service;
   private final AbstractMetaStore primaryMetastore = newPrimaryInstance(PRIMARY_NAME, URI);
-  private final FederatedMetaStore federatedMetastore = newFederatedInstance(FEDERATED_NAME, URI);
-  private @Mock Iface primaryDatabaseClient;
+  private FederatedMetaStore federatedMetastore = newFederatedInstance(FEDERATED_NAME, URI);
   private MetaStoreMapping metaStoreMappingPrimary;
   private MetaStoreMapping metaStoreMappingFederated;
+  private final List<String> mappedFederatedDatabases = Lists.newArrayList("federated_DB");
 
   @Before
   public void init() throws Exception {
-    federatedMetastore.setMappedDatabases(Lists.newArrayList("federated_DB"));
+    federatedMetastore.setMappedDatabases(mappedFederatedDatabases);
 
     metaStoreMappingPrimary = mockNewMapping(true, primaryMetastore);
     when(metaStoreMappingPrimary.getClient()).thenReturn(primaryDatabaseClient);
     when(primaryDatabaseClient.get_all_databases()).thenReturn(Lists.newArrayList("primary_db"));
     metaStoreMappingFederated = mockNewMapping(true, federatedMetastore);
+    when(metaStoreMappingFederated.getClient()).thenReturn(federatedDatabaseClient);
+    when(federatedDatabaseClient.get_databases("federated_DB")).thenReturn(mappedFederatedDatabases);
```

Comment:
This test class had to be changed quite a bit because there were many NPEs saying that there is no client found, or failed tests because the client wasn't returning anything for `get_databases`.